src/qnx/qnx-host-session.vala

```diff
@@ -0,0 +1,122 @@
+#if QNX
+namespace Frida {
+	public class QnxHostSessionBackend : Object, HostSessionBackend {
+		private QnxHostSessionProvider local_provider;
+
+		public async void start () {
+			assert (local_provider == null);
+			local_provider = new QnxHostSessionProvider ();
+			provider_available (local_provider);
+		}
+
+		public async void stop () {
+			assert (local_provider != null);
+			provider_unavailable (local_provider);
+			yield local_provider.close ();
+			local_provider = null;
+		}
+	}
+
+	public class QnxHostSessionProvider : Object, HostSessionProvider {
+		public string name {
+			get { return "Local System"; }
+		}
+
+		public ImageData? icon {
+			get { return null; }
+		}
+
+		public HostSessionProviderKind kind {
+			get { return HostSessionProviderKind.LOCAL_SYSTEM; }
+		}
+
+		private QnxHostSession host_session;
+
+		public async void close () {
+			if (host_session != null)
+				yield host_session.close ();
+			host_session = null;
+		}
+
+		public async HostSession create () throws Error {

```

Comment:
Never mind. I figured this out when it wouldn't compile. I've fixed this in the latest commit.